sgkit/stats/ld.py

```diff
@@ -61,3 +70,377 @@ def rogers_huff_r_between(gn0: ArrayLike, gn1: ArrayLike) -> float:  # pragma: n
+
+    Parameters
+    ----------
+    df
+        Dataframe containing a sparse matrix of R2 values, typically from :func:`sgkit.ld_matrix`.
+
+    Raises
+    ------
+    ValueError
+        If ``i`` and ``j`` are not sorted ascending (and in that order)
+
+    Returns
+    -------
+    A dataset containing the indexes of variants to drop, as defined by
+    :data:`sgkit.variables.ld_prune_index_to_drop_spec`.
+    """
+    args = [np.asarray(df[c]) for c in ["i", "j"]]
+    if "cmp" in df:
+        args.append(np.asarray(df["cmp"]))
+    drop = _maximal_independent_set(*args)
+    return Dataset(
+        {variables.ld_prune_index_to_drop: ("variant_indexes", np.asarray(drop))}
+    )
+
+
+def ld_prune(
+    ds: Dataset,
+    *,
+    dosage: Hashable = variables.dosage,
+    threshold: float = 0.2,
+    ld_score: Optional[Hashable] = None,
+) -> Dataset:
+    """Prune variants in linkage disequilibrium (LD).
+
+    This method uses a sparse LD matrix to find a maximally independent set (MIS)
+    of variants, and returns a dataset containing only those variants.
+
+    No information about which variants are pruned is returned by this method.
+    Consider using :func:`sgkit.ld_matrix` and :func:`sgkit.maximal_independent_set`
+    to get more insight into the variants that are pruned.
+
+    Note: This result is not a true MIS if ``ld_score`` is provided and are based on
+    minor allele frequency or anything else that is not identical for all variants.
+
+    Parameters
+    ----------
+    ds
+        Dataset containing genotype dosages.
+    dosage
+        Name of genetic dosage variable.
+        Defined by :data:`sgkit.variables.dosage_spec`.
+    threshold
+        R2 threshold below which no variant pairs will be returned. This should almost
+        always be something at least slightly above 0 to avoid the large density very
+        near zero LD present in most datasets.
+    ld_score
+        Optional name of variable to use to prioritize variant selection
+        (e.g. minor allele frequency). Defaults to None.
+        Defined by :data:`sgkit.variables.ld_score_spec`.
+
+    Returns
+    -------
+    A dataset where the variants in linkage disequilibrium have been removed.
+
+    Raises
+    ------
+    ValueError
+        If the dataset is not windowed.
+
+    Examples
+    --------
+
+    >>> import numpy as np
+    >>> import sgkit as sg
+    >>> ds = sg.simulate_genotype_call_dataset(n_variant=10, n_sample=4)
+    >>> ds.dims["variants"]
+    10
+
+    >>> # Calculate dosage
+    >>> ds["dosage"] = ds["call_genotype"].sum(dim="ploidy")
+
+    >>> # Divide into windows of size five (variants)
+    >>> ds = sg.window(ds, size=5)
+
+    >>> pruned_ds = sg.ld_prune(ds)
+    >>> pruned_ds.dims["variants"]
+    6
+    """
+    ldm = ld_matrix(ds, dosage=dosage, threshold=threshold, ld_score=ld_score)
+    mis = maximal_independent_set(ldm)
```

Comment:
What happens if the variants in the provided dataset are on different contigs?  Will the windowing prevent comparisons of variants at, say, the end of chr1 to those at the beginning of chr2?

Something that may be worth mentioning in the docs here is that this MIS algorithm will pull the entire ld_matrix data frame into memory.  The MIS steps could run separately on different contigs, but afaik there is no distributed MIS algorithm that didn't look pretty hard to do on dask.

Reply:
> Will the windowing prevent comparisons of variants at, say, the end of chr1 to those at the beginning of chr2?

Yes, since windowing is contig-aware (#372).

> Something that may be worth mentioning in the docs here is that this MIS algorithm will pull the entire ld_matrix data frame into memory.

Done. And I opened #569 for partitioning MIS by contig.